dat/generated_tables.py

```diff
@@ -171,3 +192,193 @@ def with_schema_change(case: TestCaseInfo, spark: SparkSession):
         'overwriteSchema', True).format('delta').save(
         case.delta_root)
     save_expected(case)
+
+
+@reference_table(
+    name='all_primitive_types',
+    description='Table containing all non-nested types',
+)
+def create_all_primitive_types(case: TestCaseInfo, spark: SparkSession):
+    schema = types.StructType([
+        types.StructField('utf8', types.StringType()),
+        types.StructField('int64', types.LongType()),
+        types.StructField('int32', types.IntegerType()),
+        types.StructField('int16', types.ShortType()),
+        types.StructField('int8', types.ByteType()),
+        types.StructField('float32', types.FloatType()),
+        types.StructField('float64', types.DoubleType()),
+        types.StructField('bool', types.BooleanType()),
+        types.StructField('binary', types.BinaryType()),
+        types.StructField('decimal', types.DecimalType(5, 3)),
+        types.StructField('date32', types.DateType()),
+        types.StructField('timestamp', types.TimestampType()),
+    ])
+
+    df = spark.createDataFrame([
+        (
+            str(i),
+            i,
+            i,
+            i,
+            i,
+            float(i),
+            float(i),
+            i % 2 == 0,
+            bytes(i),
+            Decimal('10.000') + i,
+            date(1970, 1, 1) + timedelta(days=i),
+            datetime(1970, 1, 1) + timedelta(hours=i)
+        )
+        for i in range(5)
+    ], schema=schema)
+
+    df.repartition(1).write.format('delta').save(case.delta_root)
+
+
+@reference_table(
+    name='nested_types',
+    description='Table containing various nested types',
+)
+def create_nested_types(case: TestCaseInfo, spark: SparkSession):
+    schema = types.StructType([
+        types.StructField(
+            'pk', types.IntegerType()
```

Comment:
chispa doesn't support ignoring sort order when comparing tables that contain map types, so we have to add a `pk` column that tests can sort on. I've added a note about this to the readme.